### ISSUE TITLE
Keep model order, not alphabetical 

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Scaffolding#283]
Scaffolder puts fields in alpha order in the views – you would never want that, you’d want the same order you defined your properties in.  You get things like First Name, Hire Date, Last Name – you’d never write a model class that way.

cc @tdykstra

**Comment by prafullbhosale** [comment]
Scaffolding retrieves the property information from `IEntityType` which returns the properties in alphabetical order. 

MVC5 scaffolding used `System.Web.DynamicData.MetaModel` and `System.Web.DynamicData.MetaTable` to get the properties in declared order. 

@divega do you know if there is any equivalent API that I can use that will keep the declared order of the properties?

**Comment by divega** [comment]
My recommendation would be to merge the EF metadata with the original reflection metadata so that you can obtain the position of members from the latter.

TL;DR: In EF Core metadata we made the intentional choice to adopt alphabetical order because it is deterministic and stable across different platforms. In EF6 and before we had had many tricky issues because we tried to preserve the reflection order, which has never been deemed deterministic and can vary depending on the platform, version of the JIT compiler, especially when there is inheritance involved. 

Of course, the fact that properties are sorted alphabetically in EF metadata doesn't mean that users don't want the declared order (or its approximation, reflection order) to be preserved. In fact, 
we have https://github.com/aspnet/EntityFramework/issues/2272 in the EF Core backlog to preserve the declared order of properties when we produce a database table schema from the entity class (e.g. in Migrations) and the same could be applied to reverse engineering.
